packages/cli/src/daemon/did-utils.ts

```diff
@@ -0,0 +1,49 @@
+import { Ed25519Provider } from "key-did-provider-ed25519"
+import * as u8a from 'uint8arrays'
+import { DID } from 'dids'
+import * as KeyDidResolver from 'key-did-resolver'
+import { Resolver } from 'did-resolver'
+import { randomBytes } from '@stablelib/random'
+
+export function makeNodeDID(seed: Uint8Array): DID {
+    const provider = makeNodeDIDProvider(seed)
+    const keyDidResolver = KeyDidResolver.getResolver()
+    const resolver = new Resolver({...keyDidResolver})
+    return new DID({ provider, resolver })
+}
+
+export function makeNodeDIDProvider(seed: Uint8Array): Ed25519Provider {
+    return new Ed25519Provider(seed)
+}
+
+export function generateSeed(): string {
+    return u8a.toString(randomBytes(32), 'base16')
+}
+
+export function generateSeedUrl(): URL {
+    const seed = generateSeed()
+    const url = `inplace:ed25519#${seed}`
```

Comment:
Ok, this explains my previous question. 
Suggestion:
Encode this as a multibase string prefixed by the `ed25519-priv` multicode: `0x1300`.

e.g.
`multibase(varint.encode(0x1300), seed)`

References:
https://github.com/multiformats/multicodec/blob/master/table.csv#L160
https://github.com/multiformats/multibase
https://www.npmjs.com/package/varint

Reply:
would keep this as a url. at least that way we can accept seeds from different locations in the future.

but yeah can multibase encode the `inplace` ones

Reply:
Is there a requirement for this is a URL? 
Using the `0x1300` only make sense if we store the actual private key, not the seed.

Reply:
I think storing seed is actually nice because composedb cli has `composedb did:generate-private-key` which we can point people to. (though it also produces a seed, not a private key)